lib/content-services/dialogs/node-lock.dialog.spec.ts

```diff
@@ -0,0 +1,160 @@
+            component.data = {
+                node: {
+                    id: 'node-id',
+                    name: 'node-name',
+                    isLocked: true,
+                    properties: {
+                        ['cm:lockType']: 'WRITE_LOCK',
+                        ['cm:expiryDate']: expiryDate
+                    }
+                }
+            };
+            fixture.detectChanges();
+        });
+
+        it('should init dialog with form inputs', () => {
+            expect(component.nodeName).toBe('node-name');
+            expect(component.form.value.isLocked).toBe(true);
+            expect(component.form.value.allowOwner).toBe(true);
+            expect(component.form.value.isTimeLock).toBe(true);
+            expect(component.form.value.time.format()).toBe(expiryDate.format());
+        });
+
+        it('should update form inputs', () => {
+            let newTime = moment();
+            component.form.controls['isLocked'].setValue(false);
+            component.form.controls['allowOwner'].setValue(false);
+            component.form.controls['isTimeLock'].setValue(false);
+            component.form.controls['time'].setValue(newTime);
+
+            expect(component.form.value.isLocked).toBe(false);
+            expect(component.form.value.allowOwner).toBe(false);
+            expect(component.form.value.isTimeLock).toBe(false);
+            expect(component.form.value.time.format()).toBe(newTime.format());
+        });
+
+        it('should submit the form and lock the node', () => {
+            spyOn(alfrescoApi.nodesApi, 'lockNode').and.returnValue(Promise.resolve({}));
+
+            component.submit();
+
+            expect(alfrescoApi.nodesApi.lockNode).toHaveBeenCalledWith(
+                'node-id',
+                {
+                    'timeToExpire': 60,
+                    'type': 'ALLOW_OWNER_CHANGES',
+                    'lifetime': 'PERSISTENT'
+                }
+            );
+        });
+
+        it('should submit the form and unlock the node', () => {
+            spyOn(alfrescoApi.nodesApi, 'unlockNode').and.returnValue(Promise.resolve({}));
+
+            component.form.controls['isLocked'].setValue(false);
+            component.submit();
+
+            expect(alfrescoApi.nodesApi.unlockNode).toHaveBeenCalledWith('node-id');
+        });
+
+        it('should call dialog to close with form data when submit is succesfluly', fakeAsync(() => {
+            const node = { entry: {} };
+            spyOn(alfrescoApi.nodesApi, 'lockNode').and.returnValue(Promise.resolve(node));
+
+            component.submit();
+            tick();
+            fixture.detectChanges();
+
+            expect(dialogRef.close).toHaveBeenCalledWith(node.entry);
+        }));
+
+        it('should handle error if submit fails', fakeAsync(() => {
+            spyOn(alfrescoApi.nodesApi, 'lockNode').and.returnValue(Promise.reject('error'));
+            spyOn(component, 'handleError').and.callFake(val => val);
```

Comment:
Method of the component under test should not be stubbed. This feature can be tested through the error output of the component.